tests/test_infer.py

```diff
@@ -2277,3 +2277,8 @@ def test_is(x, y):
 )
 def test_is_not(x, y):
     return x is not y
+
+
+def test_pair():
+    with pytest.raises(MyiaTypeError):
+        _pair(None, None)
```

Comment:
This test would be more appropriate as a test case in `test_infer` where you pass an illegal type, like an array.